### ISSUE TITLE
Can't pass a config object to the backstop command with Docker

### ISSUE THREAD

**Issue by engelfrost** [garris/BackstopJS#987]
I have a Backstop project set up where the config is exported as a function which takes an options parameter ([as described in the README](https://github.com/garris/BackstopJS#parse-a-config-file-explicitly)).

I decided to start running tests with Docker, but when I did, running the tests failed. 

```
Delegating command to Docker... docker run --rm -it --mount type=bind,source="/Users/josef/Code/backstopSanityTest",target=/src backstopjs/backstopjs:3.8.8 test "--moby=true" "--config=[object Object]" ""
```

`--config=[object Object]` is of course not a valid configuration. 

Is this an issue with the code or with the documentation?

**Comment by garris** [comment]
I suspect this would work if you were to run backstop locally -- without the docker flag.

I have a feeling that the docker implementation (in backstop) is not compatible with this approach.  My guess is that this would be pretty easy to fix tho.

**Comment by garris** [comment]
Does this work if you were to run backstop locally -- without the docker flag?

I have a feeling that the docker implementation (in backstop) is not compatible with this approach.  My guess is that this fixable if you have the time to look into it.

**Comment by garris** [comment]
Does this work if you were to run backstop locally -- without the docker flag?

I have a feeling that the docker implementation (in backstop) is not compatible with this approach.  My guess is that this fixable if you have the time to look into it.

**Comment by engelfrost** [comment]
Locally it works just fine, it's only when I add the Docker flag it fails. It's the combination of config object literal + Docker that causes the issue.

**Comment by garris** [comment]
Yup,  thanks for confirming that.  I guess to enable this the right way we would want to pass a fully rendered config to the docker instance -- which would take a bit of work.

**Comment by engelfrost** [comment]
Yes, would it be acceptable to generate a temporary config file and delete it once the test is complete?

**Comment by garris** [comment]
Sounds good to me.  Maybe create a temp file with common prefix and unique suffix name in parallel with working config file.  Something like that?
